### ISSUE TITLE
Incorrect return value when running multiple check suites 

### ISSUE THREAD

**Issue by mhidas** [ioos/compliance-checker#534]
When a single call to the checker requests multiple check suites, the final True/False return value is determined by just one of the suites, while the others are effectively ignored.

E.g. this if a file passes the CF checks:
```python
In [4]: from compliance_checker.runner import ComplianceChecker, CheckSuite

In [5]: CheckSuite.load_all_available_checkers()

In [6]: infile = 'cf_yes_acdd_no.nc'

In [7]: ComplianceChecker.run_checker(infile, ['cf'], 0, 'normal')
--------------------------------------------------------------------------------
                    The dataset scored 291 out of 291 points                    
                              during the cf check                               
--------------------------------------------------------------------------------
All tests passed!
Out[7]: (True, False)

```
... but fails ACDD:
```python
In [8]: ComplianceChecker.run_checker(infile, ['acdd'], 0, 'normal')


--------------------------------------------------------------------------------
                     The dataset scored 42 out of 60 points                     
                             during the acdd check                              
--------------------------------------------------------------------------------
                               Scoring Breakdown:                               
...
Out[8]: (False, False)
```

However, if I run both check suites at once,
```python
In [9]: ComplianceChecker.run_checker(infile, ['acdd', 'cf'], 0, 'normal')


--------------------------------------------------------------------------------
                     The dataset scored 42 out of 60 points                     
                             during the acdd check                              
--------------------------------------------------------------------------------

...

--------------------------------------------------------------------------------
                    The dataset scored 291 out of 291 points                    
                              during the cf check                               
--------------------------------------------------------------------------------
All tests passed!
Out[9]: (True, False)
```
I get the text correct output for both suites, but the final result is `True`, i.e. compliant.

Surely the correct return value in this case would be `False`?

I think the problem is in the [ComplianceChecker.stdout_output](https://github.com/ioos/compliance-checker/blob/3a696d14aab08638b3b4ebb47fc4f8cfcb2444e6/compliance_checker/runner.py#L104) method, which only returns the result `groups` for the last check suite it iterates over.

**Comment by benjwadams** [comment]
@mhidas, I've just gotten back around to revisiting some of the compliance checker results.  I'm pretty sure the above commit will fix the issues described, but I don't have any files like the one you described where all values are passing for a particular checker.  Do you still have that input file and could I use it or something similar to create a test to ensure that it works as desired?

**Comment by benjwadams** [comment]
Fixed up by combination of #644 and #652 .   Unit test and call from command line now show the appropriate behavior when running against the `ncei_gold_point_1.nc` file generated by the unit tests.  Closing.

**Comment by mhidas** [comment]
:+1: Thanks @benjwadams ! Apologies for the slow response (I've been out of the office). That looks  good to me. I just ran the latest master code on the file I used in the  examples above and it now  correctly  returns a  fail for the combined result.
